engine/src/main/java/org/terasology/logic/characters/CharacterSystem.java

```diff
@@ -73,10 +74,21 @@
     @In
     private Time time;
 
+    @In
+    private TranslationSystem translationSystem;
+
     @ReceiveEvent(components = {CharacterComponent.class})
     public void onDeath(DoDestroyEvent event, EntityRef entity) {
         CharacterComponent character = entity.getComponent(CharacterComponent.class);
-        character.controller.send(new DeathEvent());
+        DeathEvent deathEvent = new DeathEvent();
+        //Store the details of the death in the event for display on the death screen
+        deathEvent.lastDamageType = event.getDamageTypeString();
+        if (event.getInstigatorString() != null) {
+            deathEvent.lastInstigator = event.getInstigatorString();
+        } else {
+            deathEvent.lastInstigator = translationSystem.translate("${engine:menu#unknown}");
+        }
```

Comment:
Is `unknown` appropriate for no instigator?
I tested it with the fall damage and `Unknown killed you with physical damage` is not what i would expect, more like `You died to physical damage`. Are there other cases, where `unknown` fits better?